phoenix-core/src/main/java/org/apache/phoenix/calcite/CalciteUtils.java

```diff
@@ -1036,4 +1048,40 @@ public Void visitCall(RexCall call) {
             return null;
         }
     }
+
+    public static Object convertLiteral(SqlLiteral literal, PhoenixRelImplementor implementor) {
+        try {
+            final SchemaPlus rootSchema = Frameworks.createRootSchema(true);
+            final FrameworkConfig config = Frameworks.newConfigBuilder()
+                    .parserConfig(SqlParser.Config.DEFAULT)
+                    .defaultSchema(rootSchema).build();
+            Planner planner = Frameworks.getPlanner(config);
+
+            SqlParserPos POS = SqlParserPos.ZERO;
+            final SqlNodeList selectList =
+                    new SqlNodeList(
+                            Collections.singletonList(literal),
+                            SqlParserPos.ZERO);
+
+
+            String sql = new SqlSelect(POS, SqlNodeList.EMPTY, selectList, null, null, null, null,
+                    SqlNodeList.EMPTY, null, null, null).toString();
+            SqlNode sqlNode = planner.parse(sql);
+            sqlNode = planner.validate(sqlNode);
+            Project proj = (Project) (planner.rel(sqlNode).rel);
+            RexNode rex = proj.getChildExps().get(0);
+
+            Expression e = CalciteUtils.toExpression(rex, implementor);
+            ImmutableBytesWritable ptr = new ImmutableBytesWritable();
+            e = ExpressionUtil.getConstantExpression(e, ptr);
+            Object ret = e.getDataType().toObject(ptr);
+            if(ret instanceof NlsString){
```

Comment:
Yes I meant that as well. I think it's because we had originally intended to use Calcite utilities to do this. I just ran through a debugger and all inputs parse down to either a Char or a String. It is definitely covered by test cases.